### ISSUE TITLE
[bug] ReadTimeoutError from httpConnectionPool

### ISSUE THREAD

**Issue by Belpaire** [conan-io/conan#9870]
### Environment Details (include every applicable attribute)
  * windows 10, version 1909, OS build 18363.1801
  *compiler version: N/A (binary upload only)
  * Conan version: 1.38.0
  * Python version: N/A (conan windows installation)

### Steps to reproduce (Include if Applicable)
Probably any package with this size (bunch of binaries that are uncompressed about 800MB in size) will output such an error.

### Logs (Executed commands with output) (Include/Attach if Applicable)

Are you sure you want to upload '<package-name>/2021.10.26@<Suffix-name>' to '<Server-name>'? (yes/no): y
Uploading to remote '<Server-name>':
Uploading <package-name>/2021.10.26@<Suffix-name> to remote '<Server-name>'
Uploaded conanfile.py -> <package-name>/2021.10.26@<Suffix-name> [0.24k]
Uploaded conanmanifest.txt -> <package-name>/2021.10.26@<Suffix-name> [0.06k]
Uploaded conan recipe '<package-name>/2021.10.26@<Suffix-name>' to '<Server-name>': http://<URL-Name>:8082/artifactory/api/conan/conan-local
Uploading package 1/4: 01edd76db8e16db9b38c3cca44ec466a9444c388 to '<Server-name>'
Uploaded conan_package.tgz -> <package-name>/2021.10.26@<Suffix-name>:01ed [0.06k]      
Uploaded conaninfo.txt -> <package-name>/2021.10.26@<Suffix-name>:01ed [0.25k]
Uploaded conanmanifest.txt -> <package-name>/2021.10.26@<Suffix-name>:01ed [0.06k]      
Uploading package 2/4: 0a420ff5c47119e668867cdb51baff0eca1fdb68 to '<Server-name>'
Uploaded conan_package.tgz -> <package-name>/2021.10.26@<Suffix-name>:0a42 [391066.44k] 
ERROR: HTTPConnectionPool(host='<URL-Name>', port=8082): Max retries exceeded with url: /artifactory/api/conan/conan-local/v2/conans/<package-name>/2021.10.26/<Suffix-name>/revisions/2969498e68c8e43bff908838954d58f3/packages/0a420ff5c47119e668867cdb51baff0eca1fdb68/revisions/696d0a3e378e26f801fad3520a37d7a6/files/conan_package.tgz (Caused by ReadTimeoutError("HTTPConnectionPool(host='<URL-Name>', port=8082): Read timed out. (read timeout=60.0)",))
Waiting 5 seconds to retry...
Uploaded conan_package.tgz -> <package-name>/2021.10.26@<Suffix-name>:0a42 [391066.44k] 
ERROR: 
Error uploading file: conan_package.tgz, 'HTTPConnectionPool(host='<URL-Name>', port=8082): Max retries exceeded with url: /artifactory/api/conan/conan-local/v2/conans/<package-name>/2021.10.26/<Suffix-name>/revisions/2969498e68c8e43bff908838954d58f3/packages/0a420ff5c47119e668867cdb51baff0eca1fdb68/revisions/696d0a3e378e26f801fad3520a37d7a6/files/conan_package.tgz (Caused by ReadTimeoutError("HTTPConnectionPool(host='<URL-Name>', port=8082): Read timed out. (read timeout=60.0)",))'
Uploaded conaninfo.txt -> <package-name>/2021.10.26@<Suffix-name>:0a42 [0.25k]
Uploaded conanmanifest.txt -> <package-name>/2021.10.26@<Suffix-name>:0a42 [20.87k]     
Uploading package 3/4: 45b43bcd991ec89baf1eb618b595ca3227a3cf0b to '<Server-name>'
Compressing conan_package.tgz completed [0 files]
Uploaded conaninfo.txt -> <package-name>/2021.10.26@<Suffix-name>:45b4 [0.24k]
Uploaded conanmanifest.txt -> <package-name>/2021.10.26@<Suffix-name>:45b4 [0.06k]
Uploading package 4/4: 63c5d5dc6eb958dab897e143053e17f583d55879 to '<Server-name>'
Compressing conan_package.tgz completed [303 files]
Uploaded conaninfo.txt -> <package-name>/2021.10.26@<Suffix-name>:63c5 [0.24k]
Uploaded conanmanifest.txt -> <package-name>/2021.10.26@<Suffix-name>:63c5 [20.87k]
ERROR: <package-name>/2021.10.26@<Suffix-name>:0a420ff5c47119e668867cdb51baff0eca1fdb68: Upload package to '<Server-name>' failed: Execute upload again 
to retry upload the failed files: conan_package.tgz. [Remote: <Server-name>]

ERROR: Errors uploading some packages

### Additional Notes:
Probably I shouldn't have uploaded a package near to the size of 1 GB (I actually didn't realize the library provided to me was this big in size) and should probably distribute it in multiple parts instead. But I thought I'd report it here nonetheless. Note that it is correct that for 2 of the flavors I uploaded no files (As the behaviour I wanted was to only download these dlls on release builds). Also the packages seem to be uploaded correctly, but it does seem like it just wasn't able to read the response from the server.

**Comment by memsharded** [comment]
Hi @Belpaire 

It is possible to upload packages that big, but sometimes there are some annoying and difficult to debug situations like this. They tend to happen more in Windows too. There are several causes that could be happening:

- There is some memory or resources limit in the machine. Could be client side or server side.
- There could be some limitation of sizes in the server side, which one are you using?
- It might be worth trying with a modern Python 64 bits installed in your machine, then install Conan with ``pip install``
- Something that I sometimes try is to run a locally installed, in my computer, Artifactory CE (it is free), which would be typically much faster and less network limited, to try to see where in the chain it is breaking.

**Comment by Belpaire** [comment]
Hi,

Server I was uploading to is an Artifactory CE version running on windows 10. I ran Artifactory CE locally just now and I didn't receive any errors. The server I was uploading to was on-premises. So it's not an insane amount of latency that could be the issue. I didn't really do any special configuration for the Artifactory service at all. So I'm not sure what the defaults would be that could influence anything. Both Artifactories were jfrog-artifactory-cpp-ce-7.12.6-windows.zip.

**Comment by memsharded** [comment]
Maybe it could be some limitation of memory in the server, it would be worth checking the logs on the server side when it fails. Do you admin that server or can ask for those logs? Or maybe try from another computer in your LAN that you have access to the local ArtifactoryCE you are running in your machine, to try to see how much your network could be an issue.

**Comment by Belpaire** [comment]
Yes, I have access to this server so I can send you these logs. It does seem something went wrong on the artifactory side of things though, moreso than a conan issue. Is there an e-mail I could send it to?

**Comment by memsharded** [comment]
You can send private things to info@conan.io, though in general, if it doesn't expose sensitive information, or that can be hidden/removed, it is better to send it here, for reference and enabling other contributors and developers to have a look and help with it.

**Comment by Belpaire** [comment]
[artifactory-service.log](https://github.com/conan-io/conan/files/7424549/artifactory-service.log)
[artifactory-service.out.log](https://github.com/conan-io/conan/files/7424550/artifactory-service.out.log)

I only added the logging of around when the issue occurred. It seems like the server simply timed out on reading the http request.

edit: I also mistakenly said that localhost didn't have this problem, but it actually did fail again on this readtimeouterror when I uploaded it locally, so I think something in artifactory is probably choking on this 400MB file upload.

**Comment by memsharded** [comment]
I see, indeed a timeout server side. Without being an expert in Artifactory deployment I would check things:

- Possible resource limitations in the server machine.
- Check possible server other layers: is there a ISS, nginx or any other front to Artifactory? Probably there is, and this could also have its policies, limits, configuration and resources limits.
- Windows maybe is generally not great as a server for this thing, I'd try to use a Linux server if keeps being problematic

**Comment by memsharded** [comment]
Closing this ticket as inactive, I hope it was solved back then by addressing server issues, but don't hesitate to open new tickets if necessary for any further question, thanks for the feedback.
